clients/analysis/src/model/GraphDTO.ts

```diff
@@ -12,12 +12,59 @@
  *   Broadcom, Inc. - initial API and implementation
  */
 import { Location } from "./cfast";
+import { Graph } from "./Graph";
 import { Node } from "./Node";
+import { NodeDTO } from "./NodeDTO";
 
-export type GraphDTO = {
+export type EdgeFrom = number;
+export type EdgeTo = number;
+
+export class GraphDTO {
   id: number;
   programName: string;
   location: Location;
-  nodes: [number, Node][];
-  edges: [number, number[]][];
-};
+  nodes: NodeDTO[] = [];
+  nodeIdMap: Map<number, number> = new Map();
+  incomingEdges: EdgeFrom[][] = [];
+  outgoingEdges: EdgeTo[][] = [];
```

Comment:
These are needed for the CCF UI, why are we precomputing this stuff here?
The original data structure was the correct one.